presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveViewsLegacy.java

```diff
@@ -241,6 +243,40 @@ public void testSelectFromHiveViewWithoutDefaultCatalogAndSchema()
                 .containsExactly(row(1L));
     }
 
+    @Test
+    public void testTimestampHiveView() {
```

Comment:
Is the behavior exactly the same for both legacy and non-legacy?

Reply:
Yes. From what I have seen.

Reply:
can we have a common test code for them then? would it make sense?

Reply:
I would rather not, not a huge benefit imo. TestHiveViewsLegacy is not here to stay for a long time. For that reason, I originally decided to not tangle it with Coral test and just have a copy.

Reply:
> TestHiveViewsLegacy is not here to stay for a long time. 

I hope too, but the list in https://github.com/prestosql/presto/issues/5606 is only growing so far, so it seems fair to assume the legacy mode isn't going away any time soon.

Reply:
I will consider extracting common superclass as a followup.